packages/components/src/form-token-field/suggestions-list.js

```diff
@@ -2,157 +2,147 @@
  * External dependencies
  */
 import { map } from 'lodash';
-import scrollIntoView from 'dom-scroll-into-view';
+import scrollView from 'dom-scroll-into-view';
 import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { withSafeTimeout } from '@wordpress/compose';
 
-class SuggestionsList extends Component {
-	constructor() {
-		super( ...arguments );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.bindList = this.bindList.bind( this );
-	}
+function SuggestionsList( {
+	selectedIndex,
+	scrollIntoView,
+	match = '',
+	onHover = () => {},
+	onSelect = () => {},
```

Comment:
A slight optimization could be made by using lodash's `noop` here to reduce the number of functions being created, e.g.:

```js
import { map, noop } from 'lodash'
// ...
	onHover = noop,
	onSelect = noop,
```

Reply:
Premise: nit

Just wanted to add that another approach is to leave those function props as `undefined` by default, and use the optional chaining syntax when invoking them, e.g.

```js
const handHover = () => {
  // do stuff

  onHover?.();
}
```

This has the the advantage of relying solely on native APIs/syntax (plus it avoids initialising those values to their default value). But this is more of a personal preference / micro-optimisation, so we can keep the code as-is.

Reply:
I agree that using optional chaining and an `undefined` default is better than using lodash's `noop`. Thanks for pointing that out.

Reply:
Thanks for the helping me here 😅 I would prefer to keep things simple. Your comment will help others as well to understand of writing this way instead of `noop`